packages/bpk-component-dialog/src/BpkDialog.js

```diff
@@ -16,34 +16,49 @@
  * limitations under the License.
  */
 
-import PropTypes from 'prop-types';
+/* @flow */
+
 import React from 'react';
+import PropTypes from 'prop-types';
+import BpkModal, { type BpkModalProps } from 'bpk-component-modal';
+
+import { titlePropType } from './customPropTypes';
 
-import BpkModal from 'bpk-component-modal';
+export type Props = {
+  ...$Exact<BpkModalProps>,
+  dismissible: boolean,
+};
 
-const BpkDialog = props => {
+const BpkDialog = (props: Props) => {
   const { dismissible, ...rest } = props;
 
   return (
     <BpkModal
       {...rest}
-      closeOnScrimClick={dismissible}
       showHeader={dismissible}
+      closeOnScrimClick={dismissible}
       closeOnEscPressed={dismissible}
       fullScreenOnMobile={false}
     />
   );
 };
 
+const {
+  showHeader,
+  closeOnScrimClick,
+  closeOnEscPressed,
+  fullScreenOnMobile,
+  ...modalPropTypes
+} = BpkModal.propTypes;
+
 BpkDialog.propTypes = {
-  isOpen: PropTypes.bool.isRequired,
-  onClose: PropTypes.func.isRequired,
-  target: PropTypes.element,
+  ...modalPropTypes,
```

Comment:
This is inconsistent with the Flow type isn't it? Why is that?

Reply:
The title prop type? With flow types im not sure if it's possible to do conditional `if thisProp then thatProp is required`. We need an additional custom title prop type in the dialog because title is require if `dismissible: true` whereas the underlying one is for when `showTitle: true`